### PR TITLE
Allow mixed configurations using mybatis config file and type-safe configuration properties

### PR DESCRIPTION
**Pull request by kazuki43zoo**
I've tried that allow mixed configurations using mybatis config(`mybatis-config.xml`) file and type-safe configuration properties(`application.properties`).
A build is successful in my local machine (JDK 1.8 + maven wrapper).
Please review this.

### CODE REVIEW

File: mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java [hunk 1]
```diff
@@ -164,6 +167,69 @@ public void testWithDatabaseIdProvider() {
 		assertEquals("h2", this.context.getBean(SqlSessionFactory.class).getConfiguration().getDatabaseId());
 	}
 
+	@Test
+	public void testMixedConfigurations() {
+		// set up
+		{
```
Comment (eddumelendez) [inline mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java]: Can you remove brackets, please?

File: mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java [hunk 2]
```diff
@@ -164,6 +167,69 @@ public void testWithDatabaseIdProvider() {
 		assertEquals("h2", this.context.getBean(SqlSessionFactory.class).getConfiguration().getDatabaseId());
 	}
 
+	@Test
+	public void testMixedConfigurations() {
```
Comment (eddumelendez) [inline mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java]: Format the code using spring-boot eclipse formatter

File: mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java [hunk 3]
```diff
@@ -164,6 +167,69 @@ public void testWithDatabaseIdProvider() {
...
+					, MybatisInterceptorConfiguration.class
+					, DatabaseProvidersConfiguration.class
+			);
+		}
+		// run test
+		{
+			this.context.refresh();
+		}
+
+		// assert test
```
Comment (eddumelendez) [inline mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java]: remove comments

**Comment by eddumelendez** [comment]
Thanks for the commit @kazuki43zoo. I have added some comments.

File: mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java (hunk 1 above)
Comment (eddumelendez) [inline mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java]: Try to test small things or add more tests to test different combinations.

Comment (kazuki43zoo) [inline mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java]: OK. What pattern is required ? Plesae give me some hint :smile:

File: mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java (hunk 2 above)
Comment (kazuki43zoo) [inline mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java]: `mvn formatter:format` ?

File: mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java (hunk 1 above)
Comment (kazuki43zoo) [inline mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java]: I've added some test cases via 5e2c08ad20d767a940c2ecfc745a7335fbcc7702.
Is OK ?

File: mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java (hunk 2 above)
Comment (kazuki43zoo) [inline mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java]: I've formatted using https://github.com/spring-projects/spring-boot/blob/master/eclipse/eclipse-code-formatter.xml via c1e1c666601c8d8c74cceb513434b74d01f4fb09.

File: mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java (hunk 3 above)
Comment (kazuki43zoo) [inline mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java]: Remove comments via 0d3be9c7228ca1874034fcb9d6b04df947345117.

**Comment by kazuki43zoo** [comment]
@eddumelendez 

I've fixed your comments.
Please review again.

**Comment by eddumelendez** [comment]
@kazuki43zoo just format the code added.

**Comment by kazuki43zoo** [comment]
@eddumelendez 

I've formatted the adding code via 8a425a012b2bd382ee175732d599f01edb500b74.

**Comment by eddumelendez** [comment]
Merged. Thanks

**Comment by kazuki43zoo** [comment]
@eddumelendez Thanks for merging !!

**Pull request closed without merging**